lib/Dialect/FIRRTL/Transforms/SpecializeOption.cpp

```diff
@@ -89,12 +85,24 @@ struct SpecializeOptionPass
           });
         });
 
+    bool analysisPreserved = numInstances == 0;
+    circuit->walk([&](OptionOp optionOp) {
+      optionOp->erase();
```

Comment:
@seldridge , @uenoku , just wanted to make sure, this makes sense right ?
 The `OptionOp`  wouldn't be useful for any other passes downstream, even if we ignore an `InstanceChoiceOp`, when it is not specialized, and left in IR after this pass.

Reply:
This would only be safe for options that were specialized. If any `InstanceChoiceOp` is left in, e.g., if the circuit is partially instance-choice specialized, then this would create an invalid symbol reference.

Reply:
Yes, that makes sense. But since, there are no other passes that can handle the `OptionOp` maybe its okay to delete them. And this pass also errors out on partially specified instance choice. WDYT ?

Reply:
All the other passes are _supposed_ to handle these things, just it's not implemented, yet. 🫠 The concern is more about if this pass is guaranteed to specialize away everything (which we eventually don't want), then it's fine. However, if it is going to leave anything in, then it shouldn't delete everything.

Reply:
Right, for now, this pass currently doesn't leave any `InstanceChoiceOp`, it should error out if a choice is not selected. So, it should be fine to delete it.
But the first pass that needs to handle `OptionOp` is `LowerToHW`, there is no `HW` counterpart for it.

Reply:
Yep... just showing something end-to-end would be cool. FWIW, I am totally fine with `LowerToHW` erroring with "unsupported operation". That's all part of getting a feature turned on incrementally. 😉